tests/vm_networking_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/google/goexpect"
+
+	"fmt"
+
+	v12 "k8s.io/api/core/v1"
+	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/onsi/ginkgo/extensions/table"
+
+	"sync"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Networking", func() {
+	dockerTag := os.Getenv("docker_tag")
+	if dockerTag == "" {
+		dockerTag = "latest"
+	}
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+	//	virtConfig, err := kubecli.GetKubevirtClientConfig()
+	//	tests.PanicOnError(err)
+
+	var inboundVM *v1.VirtualMachine
+	var outboundVM *v1.VirtualMachine
+
+	// TODO this is not optimal, since the one test which will initiate this, will look slow
+	BeforeAll(func() {
+		tests.BeforeTestCleanup()
+
+		var wg sync.WaitGroup
+
+		createAndLogin := func() *v1.VirtualMachine {
+			vm := tests.NewRandomVMWithEphemeralDiskAndUserdata("kubevirt/cirros-registry-disk-demo:devel", "#!/bin/bash\necho 'hello'\n")
```

Comment:
do you need a VM with userdata here or would a registry disk without cloud-init be okay? I don't see the userdata being a part of the test.

Reply:
This is needed to speed up the boot process. Without the userdata cirros image will wait for it for some time until it gives up.